joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstCreator.scala

```diff
@@ -885,11 +885,15 @@ class AstCreator(filename: String, phpAst: PhpFile, fileContent: Option[String],
       val fieldIdentifier = newFieldIdentifierNode(memberNode.name, memberNode.lineNumber)
       callAst(fieldAccessNode, List(identifier, fieldIdentifier).map(Ast(_))).withRefEdges(identifier, thisParam.toList)
     } else {
-      val identifierCode = memberNode.code.replaceAll("const ", "").replaceAll("case ", "")
-      val typeFullName   = Option(memberNode.typeFullName)
-      val identifier = newIdentifierNode(memberNode.name, typeFullName.getOrElse("ANY"))
-        .code(identifierCode)
-      Ast(identifier).withRefEdge(identifier, memberNode)
+      val selfIdentifier = {
+        val name = "self"
+        val typ  = scope.getEnclosingTypeDeclTypeName
+        newIdentifierNode(name, typ.getOrElse("ANY"), typ.toList, memberNode.lineNumber).code(name)
+      }
+      val fieldIdentifier = newFieldIdentifierNode(memberNode.name, memberNode.lineNumber)
+      val code = s"self::${memberNode.code.replace("static ", "").replace("case ", "").replace("const ", "")}"
```

Comment:
I think should work, and if it does will be more concise
```suggestion
      val code = s"self::${memberNode.code.replaceAll("(static|case|const) ", "")}"
```

Reply:
Applied. Thanks